sparkle/src/components/Breadcrumbs.tsx

```diff
@@ -63,45 +87,76 @@ export function Breadcrumbs({ items, className }: BreadcrumbProps) {
             key={`breadcrumbs-${index}`}
             className="s-flex s-flex-row s-items-center s-gap-1"
           >
-            <Icon
-              visual={item.icon}
-              className="s-text-muted-foreground dark:s-text-muted-foreground-night"
-            />
             {item.label === ELLIPSIS_STRING ? (
               <DropdownMenu>
                 <DropdownMenuTrigger asChild>
-                  <Button variant="ghost" label={ELLIPSIS_STRING} />
+                  <Button
+                    variant="ghost"
+                    label={ELLIPSIS_STRING}
+                    icon={item.icon}
+                  />
                 </DropdownMenuTrigger>
 
                 <DropdownMenuContent align="start">
                   <DropdownMenuGroup>
                     {itemsHidden.map((item, index) => (
                       <DropdownMenuItem
                         key={`breadcrumbs-hidden-${index}`}
-                        href={item.href}
+                        href={isLinkItem(item) ? item.href : undefined}
+                        onClick={isButtonItem(item) ? item.onClick : undefined}
                         icon={item.icon}
                         label={item.label}
                       />
                     ))}
                   </DropdownMenuGroup>
                 </DropdownMenuContent>
               </DropdownMenu>
-            ) : item.href ? (
-              <Link
+            ) : isLinkItem(item) ? (
```

Comment:
Not related to this PR but wouldn't it be easier to read if we extract breadcrumbs item to a separate component, instead of nested ternary?  
```
 if (item.label === ELLIPSIS_STRING) {
    return (
      <DropdownMenu>
       ...
  }
  if (isLinkItem(item)) {
    return (
      <Button
   ....
}


```

and also it can have `const isLast = itemsShown.length - 1` or something it's repeating many times 😄

Reply:
Definitely not easily readable. Just refactored it 🙏🏼